steps/11_transformer_refactor/step.md

```diff
@@ -0,0 +1,158 @@
+# refactor of transformers interface
+
+Contributors: @fkiraly, 
+
+## High-level summary 
+
+### The Aim
+
+The transformers interface has grown organically and is difficult to use, interface, and extend.
+
+This proposal tries to alleviate the problem by removing mixins from transformers and using the new tag system to provide a unified and easy-to-use interface.
+
+### The proposed solution
+
+Our proposed solution follows principles used in the refactor of the forecasters:
+
+* methods are separated into outer `fit`/`transform` and inner `_fit`/`_transform` etc
+* the four mixins are removed, and replaced by tags
+* the four transformer mixins are united under a single interface
+* addition of `update` methods for transformers, to be usable with forecaster `update`
+
+The unification introduces type inhomogeneity in the `transform` method - this is addressed by allowing both `Series` and `Panel` inputs and outputs, and dropping the `Z` argument in favour of `X` (keeping the `Z` as a downwards compatible alias).
+
+## Refactor design: transformers
+
+We proceed outlining the refactor interface.
+
+### Conceptual model: types of transformers
+
+Conceptually, we change our view of transformer types. Before the refactor, these weree:
+
+* series-to-primitives
+* panel-to-tabular
+* series-to-series
+* panel-to-panel
+* series-to-panel
+
+Where series-to-panel occurs only in two places currently and not formalised as a transformer type:
+* sliding window forecasting (in `update_predict`)
+* epoching, in `sktime-neuro`
+
+We replace the old taxonomy with an input/output and rowwise yes/no taxonomy - based on what type an individual instance has, and whether `fit`/`transform` is independent per-instance or not:
+
+| `transform-input` | `transform-output` | `rowwise` | old name |
+|---|---|---|---|
+| `Series` | `Primitives` | True | series-to-primitives |
+| `Series` | `Primitives` | False | panel-to-tabular |
+| `Series` | `Series` | True | series-to-series |
+| `Series` | `Series` | False | panel-to-panel |
+| `Series` | `Panel` | True | N/A, this is new |
```

Comment:
I now understand how the above was intended. But then why do you have `panel` here?

Reply:
well, because this is the only case where an individual series gets "changed" into a panel, i.e., one series input corresponds to one panel output.

In all the other cases, you have one-to-one correspondence between individual series and/or dataframe rows, on the index level.